### PR TITLE
armcord: 3.2.4 -> 3.2.4-libwebp

### PR DESCRIPTION
**Pull request by Arjun31415**
Libwebp safe release

https://github.com/ArmCord/ArmCord/releases/tag/v3.2.4-libwebp.

## Description of changes

Updated armcord to libwep safe version. It has been tagged with the same release 3.2.4, but in github url it has a different release name (3.2.4-libwep), so a new field in the derivation was required. Can be removed for future releases. 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by NotAShelf — APPROVED** [review]
I'm not very sure about the definition to be honest, but I'm not going to argue against it

LGTM

File: pkgs/applications/networking/instant-messengers/armcord/default.nix [hunk 1]
```diff
@@ -40,19 +40,20 @@
 stdenv.mkDerivation rec {
   pname = "armcord";
   version = "3.2.4";
+  release-name="3.2.4-libwebp";
```
Comment (wegank) [inline pkgs/applications/networking/instant-messengers/armcord/default.nix:43]:
```suggestion
  version = "3.2.4-libwebp";
```

File: pkgs/applications/networking/instant-messengers/armcord/default.nix [hunk 2]
```diff
@@ -40,19 +40,20 @@
...
 
   src =
     let
       base = "https://github.com/ArmCord/ArmCord/releases/download";
     in
       {
         x86_64-linux = fetchurl {
-          url = "${base}/v${version}/ArmCord_${version}_amd64.deb";
-          hash = "sha256-IUHcDHIJeGx7QKjxl3fUFHqUfs1JdIFxesvDXt3mVw0=";
+          url = "${base}/v${release-name}/ArmCord_${version}_amd64.deb";
```
Comment (wegank) [inline pkgs/applications/networking/instant-messengers/armcord/default.nix:51]:
```suggestion
          url = "${base}/v${version}/ArmCord_${version}_amd64.deb";
```

Comment (wegank) [inline pkgs/applications/networking/instant-messengers/armcord/default.nix:51]:
```suggestion
          url = "${base}/v${version}/ArmCord_${builtins.head (lib.splitString "-" version)}_amd64.deb";
```

**Comment by NotAShelf** [comment]
please squash all commits into one

**Comment by Arjun31415** [comment]
squashed

**Review by LudovicoPiero — APPROVED** [review]

**Pull request merged**